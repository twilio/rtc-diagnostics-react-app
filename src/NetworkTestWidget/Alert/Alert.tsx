import React from 'react';
import clsx from 'clsx';
import InfoIcon from '@material-ui/icons/InfoOutlined';
import ErrorIcon from '@material-ui/icons/ErrorOutlined';
import { darken, lighten, makeStyles, Theme } from '@material-ui/core/styles';

type AlertProps = {
  children: React.ReactNode;
  variant: 'info' | 'error';
};

const useStyles = makeStyles((theme: Theme) => {
  const getBackgroundColor = theme.palette.type === 'light' ? lighten : darken;

  return {
    item: {
      display: 'flex',
      alignItems: 'center',
      padding: '0.85em',
      borderRadius: theme.shape.borderRadius,
      '&:not(:last-child)': {
        marginBottom: '0.8em',
      },
      '& svg': {
        margin: '0 0.6em 0 0.3em',
        padding: '1px',
      },
    },
    info: {
      backgroundColor: getBackgroundColor(theme.palette.info.main, 0.9),
      '& svg': {
        fill: theme.palette.info.main,
      },
    },
    error: {
      backgroundColor: getBackgroundColor(theme.palette.error.main, 0.9),
      '& svg': {
        fill: theme.palette.error.main,
      },
    },
  };
});

export default function Alert({ variant, children }: AlertProps) {
  const classes = useStyles();

  return (
    <div className={clsx(classes.item, classes[variant])}>
      {variant === 'info' && <InfoIcon />}
      {variant === 'error' && <ErrorIcon />}
      {children}
    </div>
  );
}
