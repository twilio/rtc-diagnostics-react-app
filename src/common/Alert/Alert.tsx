import React from 'react';
import clsx from 'clsx';
import CheckIcon from '@material-ui/icons/CheckCircleOutline';
import ErrorIcon from '@material-ui/icons/ErrorOutline';
import WarningIcon from '@material-ui/icons/ReportProblemOutlined';
import { darken, lighten, makeStyles, Theme } from '@material-ui/core/styles';

type AlertProps = {
  children: React.ReactNode;
  variant: 'success' | 'warning' | 'error';
};

const useStyles = makeStyles((theme: Theme) => {
  const getBackgroundColor = theme.palette.type === 'light' ? lighten : darken;

  return {
    container: {
      display: 'flex',
      padding: '0.85em',
      borderRadius: theme.shape.borderRadius,
      '&:not(:last-child)': {
        marginBottom: '0.8em',
      },
      '& svg': {
        margin: '0 0.6em 0 0.3em',
        padding: '1px',
      },
    },
    contentContainer: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
    },
    success: {
      backgroundColor: getBackgroundColor(theme.palette.success.main, 0.9),
      '& svg': {
        fill: theme.palette.success.main,
      },
    },
    warning: {
      backgroundColor: getBackgroundColor(theme.palette.warning.main, 0.9),
      '& svg': {
        fill: theme.palette.warning.main,
      },
    },
    error: {
      backgroundColor: getBackgroundColor(theme.palette.error.main, 0.9),
      '& svg': {
        fill: theme.palette.error.main,
      },
    },
  };
});

export default function Alert({ variant, children }: AlertProps) {
  const classes = useStyles();

  return (
    <div className={clsx(classes.container, classes[variant])}>
      {variant === 'success' && <CheckIcon />}
      {variant === 'warning' && <WarningIcon />}
      {variant === 'error' && <ErrorIcon />}
      <div className={classes.contentContainer}>{children}</div>
    </div>
  );
}
