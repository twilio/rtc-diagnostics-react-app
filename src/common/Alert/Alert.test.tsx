import React from 'react';
import Alert from './Alert';
import { shallow } from 'enzyme';

describe('the Alert component', () => {
  it('should render the "success" variant correctly', () => {
    const wrapper = shallow(
      <Alert variant="success">
        <span>Test Info Content</span>
      </Alert>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the "warning" variant correctly', () => {
    const wrapper = shallow(
      <Alert variant="warning">
        <span>Test Warning Content</span>
      </Alert>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the "error" variant correctly', () => {
    const wrapper = shallow(
      <Alert variant="error">
        <span>Test Error Content</span>
      </Alert>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
