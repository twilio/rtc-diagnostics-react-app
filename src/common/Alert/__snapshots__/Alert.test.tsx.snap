// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Alert component should render the "error" variant correctly 1`] = `
<div
  className="makeStyles-container-1 makeStyles-error-5"
>
  <ErrorOutlineIcon />
  <div
    className="makeStyles-contentContainer-2"
  >
    <span>
      Test Error Content
    </span>
  </div>
</div>
`;

exports[`the Alert component should render the "success" variant correctly 1`] = `
<div
  className="makeStyles-container-1 makeStyles-success-3"
>
  <CheckCircleOutlineIcon />
  <div
    className="makeStyles-contentContainer-2"
  >
    <span>
      Test Info Content
    </span>
  </div>
</div>
`;

exports[`the Alert component should render the "warning" variant correctly 1`] = `
<div
  className="makeStyles-container-1 makeStyles-warning-4"
>
  <ReportProblemOutlinedIcon />
  <div
    className="makeStyles-contentContainer-2"
  >
    <span>
      Test Warning Content
    </span>
  </div>
</div>
`;
